src/smpp_socket.erl

```diff
@@ -955,3 +976,48 @@ pp(submit_sm_resp, _) -> record_info(fields, submit_sm_resp);
 pp(unbind, _) -> record_info(fields, unbind);
 pp(unbind_resp, _) -> record_info(fields, unbind_resp);
 pp(_, _) -> no.
+
+%%%-------------------------------------------------------------------
+%%% Overload protection
+%%%-------------------------------------------------------------------
+-spec init_rate_limit(state()) -> state().
+init_rate_limit(#{req_per_sec := undefined} = State) ->
+    State;
+init_rate_limit(#{req_per_sec := Rps} = State) ->
+    % keep one slot for eventual enquire_link request
+    State#{req_per_sec := Rps - 1,
+           current_rps := {erlang:monotonic_time(second), 0}}.
+
+-spec check_limits(state()) ->
+          {ok, state()} |
+          {error, {statename(), state(), [gen_statem:action()]}}.
+check_limits(#{req_per_sec := undefined} = State) ->
+    check_in_flight(State);
+check_limits(#{req_per_sec := MaxRps, current_rps := {CurSecond, CurRate}} = State) ->
+    Now = erlang:monotonic_time(millisecond),
+    case Now div 1000 of
```

Comment:
I still think that the scheme with using `erlang:monotonic_time(second)`, proposed by the previous comment, is more effective: this function and `rate_limit_cooldown_timeout()`  before the use always transform _Now in milliseconds_ to _seconds_ by additional `Now div 1000` operation. So what's the point of  `erlang:monotonic_time(millisecond)`?

Reply:
removed timeout calculations, now explicitly getting `erlang:monotonic_time(second)`